grove/ising/ising_qaoa.py

```diff
@@ -0,0 +1,142 @@
+                      gamma angles are not provided.
+    :param connection: (Optional) connection to the QVM. Default is None.
+    :param samples: (Optional. Default=None) VQE option. Number of samples
+                    (circuit preparation and measurement) to use in operator
+                    averaging.
+    :param initial_beta: (Optional. Default=None) Initial guess for beta
+                         parameters.
+    :param initial_gamma: (Optional. Default=None) Initial guess for gamma
+                          parameters.
+    :param minimizer_kwargs: (Optional. Default=None). Minimizer optional
+                             arguments.  If None set to
+                             {'method': 'Nelder-Mead',
+                             'options': {'ftol': 1.0e-2, 'xtol': 1.0e-2,
+                                        'disp': False}
+    :param vqe_option: (Optional. Default=None). VQE optional
+                             arguments.  If None set to
+                       vqe_option = {'disp': print_fun, 'return_all': True,
+                       'samples': samples}
+
+    """
+
+    cost_operators = []
+    driver_operators = []
+    for i, j in J.keys():
+        cost_operators.append(PauliSum([PauliTerm("Z", i, J[(i, j)]) * PauliTerm("Z", j)]))
+
+    for i in range(len(h)):
+        cost_operators.append(PauliSum([PauliTerm("Z", i, h[i])]))
+
+    n_nodes = sorted([item for sublist in J.keys() for item in sublist], reverse=True)[0]
+
+    for i in range(n_nodes + 1):
+        driver_operators.append(PauliSum([PauliTerm("X", i, -1.0)]))
+
+    if connection is None:
+        connection = CXN
+
+    if minimizer_kwargs is None:
+        minimizer_kwargs = {'method': 'Nelder-Mead',
+                            'options': {'ftol': 1.0e-2, 'xtol': 1.0e-2,
+                                        'disp': False}}
+    if vqe_option is None:
+        vqe_option = {'disp': print_fun, 'return_all': True,
+                      'samples': samples}
+
+    qaoa_inst = QAOA(connection, n_nodes + 1, steps=steps, cost_ham=cost_operators,
+                     ref_hamiltonian=driver_operators, store_basis=True,
+                     rand_seed=rand_seed,
+                     init_betas=initial_beta,
+                     init_gammas=initial_gamma,
+                     minimizer=minimize,
+                     minimizer_kwargs=minimizer_kwargs,
+                     vqe_options=vqe_option)
+
+    return qaoa_inst
+
+
+def ising(h, J, num_steps=0, verbose=True):
+    """
+    Ising method initialization
+
+    :param h: External magnectic term of the Ising problem. List.
+    :param J: Interaction term of the Ising problem. Dictionary.
+    :param num_steps: (Optional.Default=2 * len(h)) Trotterization order for the
+                  QAOA algorithm.
+    :param verbose: (Optional.Default=True) Verbosity of the code.
+    """
+    if num_steps == 0:
+        num_steps = 2 * len(h)
```

Comment:
Can you motivate this choice?

Reply:
This choice is purely based on my experience with the code. It is the number of steps that seemed to work reasonably well when finding minimum values for Ising instances. Feel free to edit this number to a more suitable one that has some reasoning behind.